src/utils.jl

```diff
@@ -21,12 +21,27 @@ macro check_args(K, param, cond, desc=string(cond))
     end
 end
 
-function vec_of_vecs(X::AbstractMatrix; obsdim::Int=2)
-    @assert obsdim ∈ (1, 2) "obsdim should be 1 or 2, see docs of kernelmatrix"
-    if obsdim == 1
-        RowVecs(X)
+function deprecated_obsdim(obsdim::Union{Int,Nothing})
+    return if obsdim === nothing
+        Base.depwarn(
+            "implicit `obsdim=2` argument is deprecated and now has to be passed " *
+            "explicitly to specify that each column corresponds to one observation",
+            :vec_of_vecs,
+        )
+        2
     else
+        obsdim
```

Comment:
I found the previous version rather more readable / more easily understandable:
```suggestion
    if obsdim === nothing
        Base.depwarn(
            "implicit `obsdim=2` argument is deprecated and now has to be passed " *
            "explicitly to specify that each column corresponds to one observation",
            :vec_of_vecs,
        )
        return 2
    else
        return obsdim
```

Reply:
I have the opposite view 😄 I prefer the version in this PR much more since I think it's cleaner and easier to parse. I read it as a short form of
```suggestion
    _obsdim = if obsdim === nothing
        Base.depwarn(
            "implicit `obsdim=2` argument is deprecated and now has to be passed " *
            "explicitly to specify that each column corresponds to one observation",
            :vec_of_vecs,
        )
        2
    else
        obsdim
    end
    return _obsdim
```
which makes it more obvious that the evaluation result of the `if` expression is returned. I guess I prefer this view and use of `if` statements as something that can be evaluated and returns a specific value due to my FP/Haskell background.

Reply:
It's not a major thing but for such parts of the code I would be on the side of @st--

Reply:
I'd still prefer your suggestion (with explicit `return _obsdim`) over the `return if ...`.